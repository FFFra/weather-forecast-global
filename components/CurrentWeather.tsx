import React from 'react';
import { View, Text, Image } from 'react-native';
import { styles } from './CurrentWeather.styles';

interface CurrentWeatherProps {
    city?: string;
    time?: string;
    country?: string;
    condition?: string;
    iconUrl?: string;
    temperature?: string;
    future?: boolean;
}

export function CurrentWeather({ city, time, country, temperature, condition, iconUrl, future }: CurrentWeatherProps) {
    return (
        <View style={[styles.container, future && styles.futureContainer]}>
            <Text style={styles.city}>{future ? time : city}</Text>
            <Text style={styles.country}>{!future && country}</Text>
            {!future && <Text style={styles.temperature}>{temperature + ' °C'}</Text>}
            <Image source={{ uri: `https:${iconUrl}` }} style={styles.icon} />
            <Text style={styles.Futuretemperature}>{future && temperature + ' °C'}</Text>
            <Text style={styles.condition}>{condition}</Text>
        </View>
    );
}
