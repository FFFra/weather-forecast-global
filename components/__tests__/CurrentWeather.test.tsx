import React from 'react';
import renderer from 'react-test-renderer';
import { CurrentWeather } from '../CurrentWeather';

describe('CurrentWeather', () => {
    it('renders default correctly', () => {
        const tree = renderer
            .create(<CurrentWeather
                city="London"
                country="UK"
                temperature="20"
                condition="Sunny"
                iconUrl="https://example.com/icon.png"
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders future weather prop', () => {
        const tree = renderer
            .create(<CurrentWeather
                city="London"
                time="12:00"
                country="UK"
                temperature="20"
                condition="Sunny"
                iconUrl="https://example.com/icon.png"
                future
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
