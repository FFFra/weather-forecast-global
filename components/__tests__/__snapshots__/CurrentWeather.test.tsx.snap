// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather renders default correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "padding": 16,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 40,
        "fontWeight": "bold",
      }
    }
  >
    London
  </Text>
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 18,
      }
    }
  >
    UK
  </Text>
  <Text
    style={
      {
        "fontSize": 60,
        "fontWeight": "light",
      }
    }
  >
    20 °C
  </Text>
  <Image
    source={
      {
        "uri": "https:https://example.com/icon.png",
      }
    }
    style={
      {
        "height": 90,
        "width": 90,
      }
    }
  />
  <Text
    style={
      {
        "color": "dimgray",
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": 400,
        "marginTop": 8,
      }
    }
  >
    Sunny
  </Text>
</View>
`;

exports[`CurrentWeather renders future wheater prop 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "padding": 16,
      },
      {
        "backgroundColor": "white",
        "borderColor": "lightgrey",
        "borderRadius": 10,
        "borderWidth": 0.9,
        "elevation": 5,
        "marginLeft": 15,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3.84,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 40,
        "fontWeight": "bold",
      }
    }
  >
    12:00
  </Text>
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 18,
      }
    }
  />
  <Image
    source={
      {
        "uri": "https:https://example.com/icon.png",
      }
    }
    style={
      {
        "height": 90,
        "width": 90,
      }
    }
  />
  <Text
    style={
      {
        "color": "dimgray",
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
  >
    20 °C
  </Text>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": 400,
        "marginTop": 8,
      }
    }
  >
    Sunny
  </Text>
</View>
`;
